File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int length = nums.size();
        int l = 0;
        int r = 0;

        while (r < length){
            nums[l] = nums [r];
            while (r < length && nums[l] == nums[r]){
                r++;
            }
            l++;
        }
        return l;
    }
};

File: Data Structures & Algorithms/house-robber-ii/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1: return nums[0]

        def robHouse(nums):
            dp = [-1 for _ in range(len(nums))]

            def dfs(n):
                if n < 0: return 0
                if dp[n] != -1: return dp[n]

                dp[n] = max(dfs(n - 1), nums[n] + dfs(n - 2))
                return dp[n]

            return dfs(len(nums) - 1)

        return max(robHouse(nums[:-1]), robHouse(nums[1:]))

